addons/payment_transfer/models/payment_transaction.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import pprint
+
+from odoo import _, api, models
+from odoo.tools.float_utils import float_compare
+
+from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class TransferPaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _get_specific_rendering_values(self, processing_values):
+        self.ensure_one()
+        if self.provider != "transfer":
+            return super()._get_specific_rendering_values(processing_values)
+
+        return {
+            'currency': self.currency_id.name,
+            'tx_url': self.acquirer_id._transfer_get_redirect_action_url(),
+            **processing_values,
+        }
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        if provider != "transfer":
+            return super()._get_tx_from_feedback_data(provider, data)
+
+        reference = data.get('reference')
+        tx = self.search([('reference', '=', reference)])
+
+        if not tx or len(tx) > 1:
+            error_msg = _('received data for reference %s') % (pprint.pformat(reference))
+            if not tx:
+                error_msg += _('; no order found')
+            else:
+                error_msg += _('; multiple order found')
+            _logger.info(error_msg)
+            raise ValidationError(error_msg)
```

Comment:
There can only be one matching tx for a given ref
```suggestion
        if not tx:
            raise ValidationError("Wire Transfer: " + _(
                "no matching transaction found for reference %s", reference
            ))
```